apps/src/weblab/WebLab.js

```diff
@@ -0,0 +1,213 @@
+let webLab_ = null;
+
+let WebLab = function () {
+  this.skin = null;
+  this.level = null;
+
+  /** @type {StudioApp} */
+  this.studioApp_ = null;
+
+  this.eventHandlers = {};
+  this.Globals = {};
+  this.reportPreloadEventHandlerComplete_ = null;
+
+  consoleApi.setLogMethod(this.log.bind(this));
+  errorHandler.setLogMethod(this.log.bind(this));
+
+  // store reference to singleton
+  webLab_ = this;
+};
+
+/**
+ * Forward a log message to both logger objects.
+ * @param {?} object
+ */
+WebLab.prototype.log = function (object) {
+  this.consoleLogger_.log(object);
+};
+
+/**
+ * Inject the studioApp singleton.
+ */
+WebLab.prototype.injectStudioApp = function (studioApp) {
+  this.studioApp_ = studioApp;
+  this.studioApp_.reset = this.reset.bind(this);
+};
+
+/**
+ * Returns the global singleton. For use by Bramble host to get the Web Lab interface.
+ * Set on the window object so that Bramble host can get to it from inside its iframe.
+ */
+window.getWebLab = function () {
+  return webLab_;
+};
+
+/**
+ * Initialize this WebLab instance.  Called on page load.
+ */
+WebLab.prototype.init = function (config) {
+  if (!this.studioApp_) {
+    throw new Error("WebLab requires a StudioApp");
+  }
+
+  this.skin = config.skin;
+  this.level = config.level;
+
+  this.brambleHost = null;
+
+  if (this.level.lastAttempt) {
+    this.startSources = this.level.lastAttempt;
+  } else if (this.level.startSources && this.level.startSources.length > 0) {
+    try {
+      this.startSources = JSON.parse(this.level.startSources);
+    } catch (err) {
+      console.error("Unable to parse startSources list", err);
+    }
+  }
+
+  config.usesAssets = true;
+
+  config.makeYourOwn = false;
+  config.centerEmbedded = false;
+  config.wireframeShare = true;
+  config.noHowItWorks = true;
+
+  config.getCodeAsync = this.getCodeAsync.bind(this);
+
+  // Provide a way for us to have top pane instructions disabled by default, but
+  // able to turn them on.
+  config.showInstructionsInTopPane = true;
+  config.noInstructionsWhenCollapsed = true;
+
+  let onMount = function () {
+    this.setupReduxSubscribers(this.studioApp_.reduxStore);
+
+    // TODO: understand if we need to call studioApp
+    // Other apps call studioApp.init(). That sets up UI that is not present Web Lab (run, show code, etc) and blows up
+    // if we call it. It's not clear there's anything in there we need, although we may discover there is and need to refactor it
+    // this.studioApp_.init(config);
+  }.bind(this);
+
+  // Push initial level properties into the Redux store
+  this.studioApp_.setPageConstants(config, {
+    channelId: config.channel,
+    isProjectLevel: !!config.level.isProjectLevel,
+  });
+
+  function onUndo() {
+      this.brambleHost.undo();
+  }
+
+  function onRedo() {
+    this.brambleHost.redo();
+  }
+
+  function onShowPreview() {
+    this.brambleHost.hideTutorial();
+  }
+
+  function onShowTutorial() {
+    this.brambleHost.showTutorial();
+  }
```

Comment:
You might be able to create each of the above as arrow functions, and then not need to re-bind them to `this` in your JSX below (though I could be wrong about that)
i.e.
```
const onShowTutorial = () => this.brambleHost.showTutorial
```
You could actually also probably refrain from creating a function completely and in your JSX just have:
```
<WeblabView
  onShowTutorial={this.brambleHost.showTutorial}
/>
```